vignettes/lab_grading.Rmd

```diff
@@ -0,0 +1,395 @@
+
+
+# Grading metadata
+
+{admiral} will store a metadata data set with required variables and optional variables, the
+optional variables are purely for transparency, and will contain detailed information about
+the grading criteria. The required variables are those used by {admiral} to create the grade.
+
+## Structure of metadata data set
+
+The metadata data set has the following structure for the required variables:
+
+Variable | Scope |  Type | Example Value
+------- | -------- | ------ | -------- 
+**TERM** | Term describing the criteria applied to a particular lab test.|  Character |  "Anemia" 
+**DIRECTION** | The direction of the abnormality of a particular lab test value|   Character | "L" or "H".
+**SI_UNIT_CHECK** | Unit of lab test, to check against input data if criteria is based on absolute values. |  Character | "mmol/L"
+**VAR_CHECK** | Comma separated list of variables used in criteria, to check input data that variables exist. |  Character | "AVAL, ANRLO"
+**GRADE_CRITERIA_CODE** | Variable to hold code that creates grade based on defined criteria. |  Character |A valid case statement within a `mutate` function call
+
+The metadata data set has the following structure for the optional variables:
+
+Variable | Scope |  Type | Example Value
+------- | -------- | ------ | -------- 
+**SOC** | System Organ Class the lab test belongs to.|  Character |  "Investigations" 
+**GRADE 1** | Grade 1 criteria for lab test, normally straight from document.|   Character | ">ULN - 3.0 x ULN".
+**GRADE 2** | Grade 2 criteria for lab test, normally straight from document.|   Character | ">3.0 - 5.0 x ULN".
+**GRADE 3** | Grade 3 criteria for lab test, normally straight from document.|   Character | ">5.0 - 20.0 x ULN".
+**GRADE 4** | Grade 4 criteria for lab test, normally straight from document.|   Character | ">20.0 x ULN".
+**DEFINITION** | Definition of abnormality, normally from source document.|  Character | "A finding based on laboratory test results that indicate an increase in the level of alanine aminotransferase (ALT or SGPT) in the blood specimen.".
+**COMMENT** | Description of any decisions made by {admiral} to implement grading criteria | Character | "Take worst case and assume on anticoagulation".
+
+
+# Creating the lab grade
+
+## Mapping ADLB VAD to the TERM variable in the {admiral} metadata data set
+
+
+```{r message=FALSE}
+library(admiral)
+library(dplyr)
+library(admiral.test)
+library(stringr)
+library(tibble)
+
+data("admiral_lb")
+
+adsl <- admiral_adsl
+lb <- admiral_lb
+
+lb <- convert_blanks_to_na(lb)
+```
+```{r echo=FALSE}
+lb <- filter(lb, USUBJID %in% c("01-701-1115", "01-705-1186", "01-705-1349", "01-708-1286", "01-707-1037", "01-716-1024"))
+```
+
+Each company needs to map their lab test to a term that describes the criteria being applied.
+The list of terms defined in the {admiral} metadata to implement NCI-CTCAEv4 is below:
+
+```{r, eval=TRUE, echo=FALSE}
+dataset_vignette(
```

Comment:
When I try and build the vignette it doesn't find this atoxgr_criteria_ctcv4
![image](https://user-images.githubusercontent.com/10111024/187243760-00143de6-0ca7-4faa-bcde-627901e14c57.png).

I can't actually run this function either?

Reply:
are you in the feature branch? maybe restart R, and then `devtools::load_all()` ?

Reply:
I can get it to run with `pkgdown` but as standalone vignette it falls over :(

Reply:
You have to properly install {dmiral} from the feature branch to make it work when running the vignette. Use the "Install and Restart" button in RStudio's "Build" tab.

![image](https://user-images.githubusercontent.com/11788080/187407108-d6b9c74a-6e80-4357-881c-1169c911f95c.png)

Reply:
Thanks @thomas-neitmann